modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/service/SubscriptionDLAppHelperLocalServiceWrapper.java

```diff
@@ -136,6 +141,18 @@ protected void deactivate() {
 		}
 	}
 
+	private boolean _hasAssetDisplayPage(ServiceContext serviceContext) {
+		int displayPageType = ParamUtil.getInteger(
+			serviceContext, "displayPageType",
+			AssetDisplayPageConstants.TYPE_DEFAULT);
+
+		if (displayPageType == AssetDisplayPageConstants.TYPE_NONE) {
+			return false;
+		}
+
+		return true;
+	}
+
```

Comment:
Is this really necessary? I think this is already checked by the asset display page friendly url provider.

Reply:
The problem here is that persistence layer is not valid, because it is persisted after the notification creation.

_assetDisplayPageEntryFormProcessor.process is called after all the _dlAppService calls in com/liferay/document/library/web/internal/portlet/action/EditFileEntryMVCActionCommand.java:1384

Reply:
I see. Then this looks good.